### PR TITLE
Implement quantile

### PR DESCRIPTION
**Pull request by phofl**
We can't really test tdigest at the moment, it's not compatible with the newest NumPy and I don't want to create a new ci job for this. 

Tested it locally and it works for now

### CODE REVIEW

**Review by crusaderky — COMMENTED** [review]
Only minor notes

File: dask_expr/_collection.py [hunk 1]
```diff
@@ -1285,6 +1286,13 @@ def rename(self, index):
             return new_collection(expr.RenameSeries(self.expr, index))
         raise NotImplementedError(f"passing index={type(index)} is not supported")
 
+    def quantile(self, q=0.5, method="default"):
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask_expr/_collection.py:1395]: Yep good point, but I’d rather do this not with all the other changes

File: dask_expr/_collection.py [hunk 2]
```diff
@@ -1285,6 +1286,13 @@ def rename(self, index):
...
+        assert pd.api.types.is_numeric_dtype(self.dtype), self.dtype
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask_expr/_collection.py:1396]: TypeErrros are caught by the getattr magic and cause recursion errors…

File: dask_expr/_quantile.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            )
+        else:
+            return SeriesQuantileDask(frame, self.operand("q"), self.operand("method"))
+
+
+class SeriesQuantileTdigest(SeriesQuantile):
+    @functools.cached_property
+    def _meta(self):
+        import_required(
+            "crick", "crick is a required dependency for using the t-digest method."
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask_expr/_quantile.py:80]: Good ping, this was actually copied 😂

File: dask_expr/_collection.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (phofl) [reply dask_expr/_collection.py:1396]: Sorry that's incorrect, mixed them up with AttributeErrors

File: dask_expr/tests/test_collection.py [hunk 4]
```diff
@@ -1669,6 +1669,13 @@ def test_quantile(df):
         lib.Series([19.0, 79.0], index=[0.2, 0.8]),
     )
 
+    with pytest.raises(AssertionError):
+        df.x.quantile(q=[]).compute()
```
Comment (crusaderky) [inline dask_expr/tests/test_collection.py:1673]: pandas returns an empty series here. Doubt we care to diverge on such an edge case though.

**Review by crusaderky — APPROVED** [review]

File: dask_expr/tests/test_collection.py (hunk 4 above)
Comment (phofl) [reply dask_expr/tests/test_collection.py:1673]: Yeah I thought this was kind of pointless, but I might put up a follow up anyway
